app/scripts/modules/amazon/src/loadBalancer/configure/network/NLBAdvancedSettings.tsx

```diff
@@ -1,38 +1,44 @@
 import React from 'react';
-import { Field, FormikProps } from 'formik';
+import { FormikProps } from 'formik';
+import { every } from 'lodash';
 
-import { HelpField } from '@spinnaker/core';
+import { HelpField, FormikFormField, CheckboxInput } from '@spinnaker/core';
 
 import { IAmazonNetworkLoadBalancerUpsertCommand } from 'amazon/domain';
 
 export interface INLBAdvancedSettingsProps {
   formik: FormikProps<IAmazonNetworkLoadBalancerUpsertCommand>;
 }
 
-export class NLBAdvancedSettings extends React.Component<INLBAdvancedSettingsProps> {
-  public render() {
-    const { values } = this.props.formik;
-    return (
-      <div className="form-group">
-        <div className="col-md-3 sm-label-right">
-          <b>Protection</b> <HelpField id="loadBalancer.advancedSettings.deletionProtection" />
-        </div>
-        <div className="col-md-7 checkbox">
-          <label>
-            <Field type="checkbox" name="deletionProtection" checked={values.deletionProtection} />
-            Enable deletion protection
-          </label>
-        </div>
-        <div className="col-md-3 sm-label-right">
-          <b>Cross-Zone Load Balancing</b> <HelpField id="loadBalancer.advancedSettings.loadBalancingCrossZone" />
-        </div>
-        <div className="col-md-7 checkbox">
-          <label>
-            <Field type="checkbox" name="loadBalancingCrossZone" checked={values.loadBalancingCrossZone} />
-            Cross-Zone Load Balancing
-          </label>
-        </div>
-      </div>
-    );
-  }
-}
+export const NLBAdvancedSettings = React.forwardRef<HTMLDivElement, INLBAdvancedSettingsProps>((props, ref) => {
+  const { formik } = props;
+  /** NLBs can only be dualstacked if they are external and all targets are IPs. This will remain true until NLBs support security groups. */
+  const allIpTargets = every(formik.values.targetGroups, { targetType: 'ip' });
```

Comment:
maybe refactor similar to the comment above passing in `allIpTargets` and `showDualstack` props instead of the entire `formik` bag